server/shared/src/main/scala/org/http4s/server/middleware/BracketRequestResponse.scala

```diff
@@ -179,25 +179,20 @@ object BracketRequestResponse {
   )(release: (A, Outcome[F, Throwable, Unit]) => F[Unit])(implicit
       F: MonadCancelThrow[F]
   ): Kleisli[F, ContextRequest[F, A], Response[F]] => Kleisli[F, Request[F], Response[F]] =
-    (contextService: Kleisli[F, ContextRequest[F, A], Response[F]]) =>
-      Kleisli((request: Request[F]) =>
-        acquire.flatMap((a: A) =>
-          contextService
-            .run(ContextRequest(a, request))
-            .map(response =>
-              response.copy(body =
-                response.body.onFinalizeCaseWeak(ec => release(a, exitCaseToOutcome(ec)))
-              )
-            )
-            .guaranteeCase {
-              case Outcome.Succeeded(_) =>
-                F.unit
-              case otherwise =>
-                release(a, otherwise.void)
-
-            }
-        )
-      )
+    (contextService: Kleisli[F, ContextRequest[F, A], Response[F]]) => {
+      def withResource(request: Request[F])(a: A): F[Response[F]] =
```

Comment:
The `withResource` uses the `contextService`, so it has to be there.

Reply:
`withResource` depends on the `contextService`, so it would not be same function. We could write a `def releaseResponse(fres: F[Response[F]]): F[Response[F]]`, but that would not change much more.

Reply:
`contextService` could be placed in a contract of function, ain't it?

Reply:
It could, yes. What would be the advantage of extracting that function to the top-level?